solutions/usaco-1181.mdx

```diff
@@ -101,5 +101,65 @@ int main() {
 ```
         
 </CPPSection>
+
+<PySection>
+
+```python
+def check(same_lvl: int, corn: [int], hunger_lvl: [int]) -> bool:
+	# check if proposed corn is achievable
+	hunger_lvl[0] -= corn[0]
+	hunger_lvl[-1] -= corn[-1]
+
+	for i in range(1, len(hunger_lvl)-1):
+		sum_of_corn = corn[i-1] + corn[i]
+		hunger_lvl[i]-= sum_of_corn
+
+	return len(set(hunger_lvl)) == 1 and same_lvl >= 0 and min(corn) >= 0
+
+def draught() -> int:
```

Comment:
does this have to be in a function? ik python runs a bit faster when in a function

Reply:
changed